### PR TITLE
implement PUT /webhooks/:webhook_id

### LINKED ISSUE #312: PUT /webhooks/{webhook_id}

### PR DESCRIPTION
**Pull request by syncush**
closes #312

### CODE REVIEW

File: src/webhooks/models/webhookManager.js [hunk 1]
```diff
@@ -7,29 +7,49 @@ const webhookDefaultValues = {
...
+
+module.exports = {
+    getAllWebhooks,
+    getWebhook,
+    createWebhook,
+    deleteWebhook,
+    updateWebhook
+};
+
+function generateError(stautsCode, message) {
```
Comment (NivLipetz) [inline src/webhooks/models/webhookManager.js:51]: this is a very generic function that we also have in `processorsManager`, lets find a better place for it

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 2]
```diff
@@ -87,6 +87,95 @@ describe('Webhooks api', function () {
                 expect(deleteWebhookResponse.statusCode).to.equal(204);
             });
         });
+        describe('PATCH /v1/webhooks/:webhook_id', function() {
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:90]:
```suggestion
        describe('PUT /v1/webhooks/:webhook_id', function() {
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 3]
```diff
@@ -162,6 +251,16 @@ describe('Webhooks api', function () {
 
                 const response = await webhookRequestSender.deleteWebhook(badWebhookValue);
 
+                expect(response.statusCode).to.equal(400);
+            });
+        });
+        describe('PATCH /v1/webhooks/:webhook_id', function() {
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:257]:
```suggestion
        describe('PUT /v1/webhooks/:webhook_id', function() {
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 4]
```diff
@@ -174,6 +273,16 @@ describe('Webhooks api', function () {
 
                 const response = await webhookRequestSender.getWebhook(notExistingWebhookId);
 
+                expect(response.statusCode).to.equal(404);
+            });
+        });
+        describe('PATCH /v1/webhooks/:webhook_id', function () {
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:279]:
```suggestion
        describe('PUT /v1/webhooks/:webhook_id', function () {
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 5]
```diff
@@ -87,6 +87,95 @@ describe('Webhooks api', function () {
...
+        describe('PATCH /v1/webhooks/:webhook_id', function() {
+            it('Insert a webhook -> update global -> ensure it is updated', async function() {
+                const webhook = generateWebhook();
+                const updatedWebhook = { ...webhook, global: !webhook.global };
+
+                const insertResponse = await webhookRequestSender.createWebhook(webhook);
+                expect(insertResponse.statusCode).to.equal(201);
+
+                const { body: { id } } = insertResponse;
+                const patchResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:99]:
```suggestion
                const updateResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 6]
```diff
@@ -87,6 +87,95 @@ describe('Webhooks api', function () {
...
+            });
+            it('Insert a webhook -> update url -> ensure it is updated', async function() {
+                const webhook = generateWebhook();
+                const updatedWebhook = { ...webhook, url: `${webhook}/new/path` };
+
+                const insertResponse = await webhookRequestSender.createWebhook(webhook);
+                expect(insertResponse.statusCode).to.equal(201);
+
+                const { body: { id } } = insertResponse;
+                const patchResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:115]:
```suggestion
                const updateResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 7]
```diff
@@ -87,6 +87,95 @@ describe('Webhooks api', function () {
...
+            it('Insert a webhook -> update format_type -> ensure it is updated', async function() {
+                const webhook = generateWebhook();
+                const updatedFormatType = EVENT_FORMAT_TYPES.filter(format => format !== webhook.format_type)[0];
+                const updatedWebhook = { ...webhook, format_type: updatedFormatType };
+
+                const insertResponse = await webhookRequestSender.createWebhook(webhook);
+                expect(insertResponse.statusCode).to.equal(201);
+
+                const { body: { id } } = insertResponse;
+                const patchResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:132]:
```suggestion
                const updateResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 8]
```diff
@@ -87,6 +87,95 @@ describe('Webhooks api', function () {
...
+            });
+            it('Insert a webhook -> update name -> ensure it is updated', async function() {
+                const webhook = generateWebhook();
+                const updatedWebhook = { ...webhook, name: webhook.name + ' added text' };
+
+                const insertResponse = await webhookRequestSender.createWebhook(webhook);
+                expect(insertResponse.statusCode).to.equal(201);
+
+                const { body: { id } } = insertResponse;
+                const patchResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:148]:
```suggestion
                const updateResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```

File: tests/integration-tests/webhooks/webhooks-test.js [hunk 9]
```diff
@@ -87,6 +87,95 @@ describe('Webhooks api', function () {
...
+                const updatedWebhook = {
+                    ...webhook,
+                    events: [WEBHOOK_EVENT_TYPE_API_FAILURE, WEBHOOK_EVENT_TYPE_FAILED]
+                };
+
+                const insertResponse = await webhookRequestSender.createWebhook(webhook);
+                expect(insertResponse.statusCode).to.equal(201);
+
+                const { body: { id } } = insertResponse;
+                const patchResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```
Comment (NivLipetz) [inline tests/integration-tests/webhooks/webhooks-test.js:170]:
```suggestion
                const updateResponse = await webhookRequestSender.updateWebhook(id, updatedWebhook);
```

File: src/webhooks/models/webhookManager.js (hunk 1 above)
Comment (syncush) [reply src/webhooks/models/webhookManager.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 2 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 3 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 4 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 5 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 6 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 7 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 8 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

File: tests/integration-tests/webhooks/webhooks-test.js (hunk 9 above)
Comment (syncush) [reply tests/integration-tests/webhooks/webhooks-test.js]: done

**Pull request merged**